src/components/alert/alert.test.ts

```diff
@@ -1,33 +1,31 @@
 import { expect, fixture, html, waitUntil } from '@open-wc/testing';
 import sinon from 'sinon';
-
-import '../../../dist/shoelace.js';
 import type SlAlert from './alert';
 
 describe('<sl-alert>', () => {
   it('should be visible with the open attribute', async () => {
     const el = await fixture<SlAlert>(html` <sl-alert open>I am an alert</sl-alert> `);
-    const base = el.shadowRoot?.querySelector('[part="base"]') as HTMLElement;
+    const base = el.shadowRoot!.querySelector<HTMLElement>('[part="base"]')!;
 
     expect(base.hidden).to.be.false;
   });
 
   it('should not be visible without the open attribute', async () => {
     const el = await fixture<SlAlert>(html` <sl-alert>I am an alert</sl-alert> `);
-    const base = el.shadowRoot?.querySelector('[part="base"]') as HTMLElement;
+    const base = el.shadowRoot!.querySelector<HTMLElement>('[part="base"]')!;
 
     expect(base.hidden).to.be.true;
   });
 
   it('should emit sl-show and sl-after-show when calling show()', async () => {
     const el = await fixture<SlAlert>(html` <sl-alert>I am an alert</sl-alert> `);
-    const base = el.shadowRoot?.querySelector('[part="base"]') as HTMLElement;
+    const base = el.shadowRoot!.querySelector<HTMLElement>('[part="base"]')!;
     const showHandler = sinon.spy();
     const afterShowHandler = sinon.spy();
 
     el.addEventListener('sl-show', showHandler);
     el.addEventListener('sl-after-show', afterShowHandler);
-    el.show();
+    void el.show();
```

Comment:
Here's my argument in favor. If I see `el.show()` there's no way to know if it's sync or async. Obviously if there's a `await` then I know that it is async and the code will behave synchronously, but if there's no `await`, then it's not obvious. So this rule enforces that async methods are explicitly declared when you don't want to await on the results. Sure it does add a few extra characters, but it also helps find bugs when you are forgetting to await a promise because you forgot that a method is actually async.

I'm optimizing for the readability of the code and explicitly voiding the promise (once you know what that means) is instantly clear what is happening.

Functionally this doesn't do anything for runtime or TypeScript, it's purely for readability and ensuring that promises are handled intentionally and not accidentally creating bugs.

Reply:
Fair point, although there are plenty of examples in the code where non-library methods are called and void isn't present. I seldom see this in the wild and it feels pretty uncomfortable and awkward to add. Can we pretty please relax this one? 🙏

Reply:
Yup of course! Turns out this is actually one of the recommended rules from the TypeScript ESLint plugin. But yup can totally disable and revert!